### PR TITLE
Add takeUntilLoaded function for Flow<Data<T>>

### PR DESCRIPTION
**Pull request by PavloRatushnyi**
- Add `takeUntilLoaded` function for `Flow<Data<T>>`
- Extract tests for `filerWhileLoading` from `ExtractContentKtTest` to `LoadingExtensionsKtTest`
- Add test for `takeUntilLoaded`
- Fix package of `FlowTestKt`

### CODE REVIEW

File: flow-extensions/src/test/kotlin/com/revolut/flowdata/extensions/LoadingExtensionsKtTest.kt [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+    @Test
+    fun `Error after loading`() = runTest {
+        val error = IllegalStateException()
+
+        flowOf(
+            Data("A", null, loading = true),
+            Data("A", error, loading = false)
+        ).filterWhileLoading().extractContent().test {
+            Assertions.assertEquals(error, awaitError())
```
Comment (yatsinar) [inline flow-extensions/src/test/kotlin/com/revolut/flowdata/extensions/LoadingExtensionsKtTest.kt:49]: nit: static imports pls

File: flow-extensions/src/test/kotlin/com/revolut/flowdata/extensions/LoadingExtensionsKtTest.kt [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+    //endregion
+
+    //region takeUntilLoaded
+
+    @Test
+    fun `Loaded item is returned and terminates the stream`() = runTest {
+        flowOf(
+            Data("A", null, loading = true),
+            Data("B", null, loading = false),
```
Comment (yatsinar) [inline flow-extensions/src/test/kotlin/com/revolut/flowdata/extensions/LoadingExtensionsKtTest.kt:74]: pls add a case for 
```
flowOf(Data("A", null, loading = false))
```
should emit Data("A", null, loading = false) and terminate right away.
(This is when memory is not empty and forceReload = false.

File: flow-extensions/src/test/kotlin/com/revolut/flowdata/extensions/LoadingExtensionsKtTest.kt [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+            Data(null, error, loading = false)
+        ).filterWhileLoading().extractContent().test {
+            Assertions.assertEquals(error, awaitError())
+        }
+    }
+
+    //endregion
+
+    //region takeUntilLoaded
+
```
Comment (yatsinar) [inline flow-extensions/src/test/kotlin/com/revolut/flowdata/extensions/LoadingExtensionsKtTest.kt:69]: Also please add cases for when errors happen at various points

**Review by yatsinar — CHANGES_REQUESTED** [review]

File: flow-extensions/src/test/kotlin/com/revolut/flowdata/extensions/LoadingExtensionsKtTest.kt (hunk 2 above)
Comment (PavloRatushnyi) [reply flow-extensions/src/test/kotlin/com/revolut/flowdata/extensions/LoadingExtensionsKtTest.kt]: makes sense, thanks, added

File: flow-extensions/src/test/kotlin/com/revolut/flowdata/extensions/LoadingExtensionsKtTest.kt [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    @Test
+    fun `Loaded item is returned and terminates the stream`() = runTest {
+        flowOf(
+            Data("A", null, loading = true),
+            Data("B", null, loading = false),
+            Data("C", null, loading = true),
+        ).takeUntilLoaded().test {
+            Assertions.assertEquals(Data("A", null, loading = true), awaitItem())
```
[the remark this answers is not in the document]
Comment (PavloRatushnyi) [reply flow-extensions/src/test/kotlin/com/revolut/flowdata/extensions/LoadingExtensionsKtTest.kt]: done

File: flow-extensions/src/test/kotlin/com/revolut/flowdata/extensions/LoadingExtensionsKtTest.kt (hunk 3 above)
Comment (PavloRatushnyi) [reply flow-extensions/src/test/kotlin/com/revolut/flowdata/extensions/LoadingExtensionsKtTest.kt:66]: done

**Review by yatsinar — APPROVED** [review]

**Pull request merged**